src/helpers/performance.js

```diff
@@ -0,0 +1,44 @@
+const performance = typeof window !== "undefined" ? window.performance : null;
+
+export const hasHighPrecision = Boolean(performance);
+
+// Safari doesn't expose the performance API, so make a wrapper API and expose a
+// lower precision form of it if not available.
+let clear = (perfMark) => {
+  performance.clearMarks(perfMark);
+  performance.clearMeasures(perfMark);
+};
+let internalMark = performance?.mark.bind(performance);
+let internalMeasure = (perfMark) => {
+  const { duration } = performance.measure(null, perfMark);
+  clear(perfMark);
+  return { time: duration, hasHighPrecision };
+};
+if (!hasHighPrecision) {
```

Comment:
This is some annoyingly obscure looking code, but unfortunately the performance API isn't consistently available. This exposes a `mark` and `measure` function to do simple timers (the perf api offers a lot more complex stuff, but start/stop seems sufficient for most cases), so this checks for if it's available and provides a shitty but functional timer stub if not.